### PR TITLE
Nblib force calculator impl

### PR DESCRIPTION
**Pull request by sekelle**

### CODE REVIEW

File: src/gromacs/nblib/forcecalculator.h [hunk 1]
```diff
@@ -48,35 +48,49 @@
...
+{
+    Geometric = 0,
+    Count     = 1
+};
 
 class ForceCalculator
 {
 public:
-
     // TODO: Depend on simulationState
```
Comment (ejjordan) [inline src/gromacs/nblib/forcecalculator.h:65]: Any idea what this means?

**Review by ejjordan — APPROVED** [review]
Even though this is not working, having all the setup code for nbnxm data structures in a (seemingly) proper state seems useful at this stage. Better to have a broken thing we can iterate on than be in limbo.

**Review by kanduri — APPROVED** [review]

File: src/gromacs/nblib/nbkerneloptions.cpp [hunk 2]
```diff
@@ -279,29 +88,29 @@ expandSimdOptionAndPushBack(const NBKernelOptions        &options,
...
-    // setupAndRunInstance(system, optionsList[0], printTimings);
-}
+//void nbKernel(NBKernelSystem        &system,
+//              const NBKernelOptions &options,
+//              const bool            &printTimings)
+//{
+//    // We don't want to call gmx_omp_nthreads_init(), so we init what we need
+//    gmx_omp_nthreads_set(emntPairsearch, options.numThreads);
+//    gmx_omp_nthreads_set(emntNonbonded, options.numThreads);
+//
```
Comment (kanduri) [inline src/gromacs/nblib/nbkerneloptions.cpp:98]: Perhaps we should quickly handle this commented code in the forthcoming PR.

File: src/gromacs/nblib/tests/topology.cpp [hunk 3]
```diff
@@ -221,26 +222,32 @@ TEST(NBlibTest, TopologyHasNonbondedParameters)
...
+//    const std::vector<int> test           = watersTopology.getAtomInfoAllVdw();
+//    std::vector<int>       ref;
+//    ref.resize(watersTopology.numAtoms());
+//    for (size_t atomI = 0; atomI < ref.size(); atomI++)
+//    {
+//        SET_CGINFO_HAS_VDW(ref[atomI]);
+//    }
+//    EXPECT_EQ(ref, test);
+//}
+
```
Comment (kanduri) [inline src/gromacs/nblib/tests/topology.cpp:250]: This too! If responsibilities have moved, then this test should also be updated.

**Pull request merged**